File: Programmers/위장.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> m;
    for(auto cloth : clothes) {
        m[cloth[1]]++; // 의상 종류 갯수 cnt
    }
        
    for(auto i = m.begin(); i != m.end(); i++) {
        answer *= (i->second+1); // 해당 의상 종류를 입지 않는 경우를 생각해 +1
    }
    return answer-1;
}

File: Programmers/위장.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>();
        for(String[] cloth : clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 0)+1);
        }
        
        for(String m : map.keySet()) {
            answer *= map.get(m)+1;
        }
        
        return answer-1;
    }
}